### ISSUE TITLE
envs config

### ISSUE THREAD

**Issue by matthewp** [stealjs/steal#454]
# Purpose

There is a lot of configuration that is specific to an environment.  Currently the functionality for loading production bundles is controlled by `System.env` being equal to `production`. This means you can't use System.env for other purposes, such as to distinguish when the loader is building or when the loader is being used in Node for server side rendering.

**envs** config is meant to decouple the environment from bundles configuration.

You can think of `System.env` as configurable a type of loader you have:

* `development`: Loading an app in development in the browser.
* `production`: Loading an app in production in the browser.
* `server`: Running in node for server-side rendering
* `build`: Running in node for building.
* `nw`: Running in NW.js

# System.envs

Provides config that is applied after `@config` has loaded depending on the `System.env` chosen.

```json
{
  "envs": {
    "server": {
      "map": {
        "jquery": {
          "socketio": "@empty"
        }
      },
      "meta": {
        "jquery": {
          "format": "global",
          "deps": ["can/util/vdom/vdom"]
        }
      }
    },
    "staging": {
      "baseURL": "http://somecdn.com/pmo/"
    }
  }
}
```

Note that since `envs` is applied after the config has loaded it can be used as a substitute for a `system-config` for most purposes.

# envMap

It is possible to have multiple environments, such as __production__ and __server__, so I propose a read-only object called `System.envMap` that includes the environments the application is running in.

System.env would be a comma separated list of environments, whenever it changed envMap would be updated.  For example:

```js
System.config({
  env: "production,server"
});

System.envMap // -> { production: true, server: true }
```

**Comment by matthewp** [comment]
# loadBundles

This replaces the purpose previously given to `System.env = "production"`.

**Comment by matthewp** [comment]
Re-implementing this to remove the envs map. The new API is documented in: https://github.com/donejs/donejs/issues/57

**Comment by matthewp** [comment]
Also need to set the default env to "window-development".

**Comment by matthewp** [comment]
Closed by #437
